src/providers/node.rs

```diff
@@ -50,20 +58,28 @@ impl Provider for NodeProvider {
     }
 
     fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
-        if NodeProvider::get_package_manager(app)? == "yarn" {
-            return Ok(Some(InstallPhase::new(
-                "yarn install --production=false --frozen-lockfile".to_string(),
-            )));
-        }
-        if app.includes_file("package-lock.json") {
-            return Ok(Some(InstallPhase::new("npm ci".to_string())));
+        let mut install_cmd: InstallPhase;
+        if NodeProvider::get_package_manager(app)? == "pnpm" {
+            install_cmd = InstallPhase::new("pnpm i --frozen-lockfile".to_string());
+            install_cmd.add_file_dependency("pnpm-lock.yaml".to_string());
+        } else if NodeProvider::get_package_manager(app)? == "yarn" {
+            install_cmd = InstallPhase::new("yarn install --frozen-lockfile".to_string());
+            install_cmd.add_file_dependency("yarn.lock".to_string());
+        } else if app.includes_file("package-lock.json") {
+            install_cmd = InstallPhase::new("npm ci".to_string());
+            install_cmd.add_file_dependency("package-lock.json".to_string());
+        } else {
+            install_cmd = InstallPhase::new("npm i".to_string());
         }
-        Ok(Some(InstallPhase::new("npm install".to_string())))
+        install_cmd.add_file_dependency("package.json".to_string());
```

Comment:
These file dependencies used to be here but I removed them since it breaks some apps since some apps run post-install scripts from something like `bin/scripts`. It is unfortunate but we might be able to add it back in the future with some more advanced detection.

For now I think it should be removed :(

Reply:
Cool, removed.